# microbial-genetic-variation

## Data
Table S1. The meta information of all metagenomic sequencing data, including geographical locations and sources.

Table S2. The sources of available reference databases.

Table S3. The meta information of 18292 metagenomic sequencing data used for metagenomic profiling, including locations and sources.
> To investigate the gut microbial genetic variations in populations, we profiled 8,584 Chinese and 9,708 non-Chinese metagenomic sequencing data (Table S3) with microbial SNPs and SVs.

Table SX. The meta information of metagenomic sequencing data in the didcovery cohort and the replication cohort.
> This study involved 7,942 individuals from four independent datasets, including CHN1 (n=2,762), CHN2 (n=1,680), Non-CHN1 (n=2,449), and Non-CHN2 (n=601).
>
> To explore the gut microbial differences between the Chinese and non-Chinese poputions, we divided them into a discovery cohort (CHN1 and Non-CHN1) and a replication cohort (CHN2 and Non-CHN2).



